src/sparseml/framework/info.py

```diff
@@ -0,0 +1,299 @@
+        description=(
+            "True if training/editing a model is available on the system "
+            "for the given framework, False otherwise."
+        ),
+    )
+    sparsification_available: bool = Field(
+        default=False,
+        title="sparsification_available",
+        description=(
+            "True if sparsifying a model is available on the system "
+            "for the given framework, False otherwise."
+        ),
+    )
+    exporting_onnx_available: bool = Field(
+        default=False,
+        title="exporting_onnx_available",
+        description=(
+            "True if exporting a model in the ONNX format is available on the system "
+            "for the given framework, False otherwise."
+        ),
+    )
+    inference_available: bool = Field(
+        default=False,
+        title="inference_available",
+        description=(
+            "True if inference for a model is available on the system "
+            "for the given framework, False otherwise."
+        ),
+    )
+
+
+def framework_info(framework: Any) -> FrameworkInfo:
+    """
+    Detect the information for the given ML framework such as package versions,
+    availability for core actions such as training and inference,
+    sparsification support, and inference provider support.
+
+    :param framework: The item to detect the ML framework for.
+        See :func:`detect_framework` for more information.
+    :type framework: Any
+    :return: The framework info for the given framework
+    :rtype: FrameworkInfo
+    """
+    _LOGGER.debug("getting system info for framework %s", framework)
+    info: FrameworkInfo = execute_in_sparseml_framework(framework, "framework_info")
+    _LOGGER.info("retrieved system info for framework %s: %s", framework, info)
+
+    return info
+
+
+def save_framework_info(framework: Any, path: Optional[str] = None):
+    """
+    Save the framework info for a given framework.
+    If path is provided, will save to a json file at that path.
+    If path is not provided, will print out the info.
+
+    :param framework: The item to detect the ML framework for.
+        See :func:`detect_framework` for more information.
+    :type framework: Any
+    :param path: The path, if any, to save the info to in json format.
+        If not provided will print out the info.
+    :type path: Optional[str]
+    """
+    _LOGGER.debug(
+        "saving framework info for framework %s to %s",
+        framework,
+        path if path else "sys.out",
+    )
+    info = (
+        framework_info(framework)
+        if not isinstance(framework, FrameworkInfo)
+        else framework
+    )
+
+    if path:
+        path = clean_path(path)
+        create_parent_dirs(path)
+
+        with open(path, "w") as file:
+            file.write(info.json())
+
+        _LOGGER.info(
+            "saved framework info for framework %s in file at %s", framework, path
+        ),
+    else:
+        print(info.json())
```

Comment:
While functionality wise this works would it be worth using pretty print so the output is more readable?